public/components/palantir/index.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './index.scss';
+
+import {
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiPopoverTitle,
+  EuiText,
+  EuiPopoverFooter,
+  EuiBadge,
+} from '@elastic/eui';
+import React, { Children, isValidElement, useState } from 'react';
+import { Palantir as PalantirInput } from '../../types';
+import { getPalantirRegistry } from '../../services';
+
+// TODO: type content
+// ?? what is the content going to be? does it change from anchor to anchor
+// ?? what is the anchor point? will assume character based
+export const Palantir = ({
+  children,
+  key,
+  input,
+  onSubmit,
+}: {
+  children: React.ReactNode;
+  key?: string;
+  input?: PalantirInput;
+  onSubmit?: () => void;
+}) => {
+  const registry = getPalantirRegistry();
+  const [isVisible, setIsVisible] = useState(false);
+
+  // TODO: error handling for bad implemention from plugins or just don't render?
+  const onAnchorClick = () => setIsVisible((visible) => !visible);
+  const onSubmitClick = async (palantirInput: PalantirInput) => {
```

Comment:
good catch. yeah i can clean this up. as you can see this can is kind of showcasing the original though that plugin developers can provide their onSubmit function and originally i was implementing utilizing this plugin to call to pass in an onSubmit function that executing and at the time it was making an async execution.